user/languages/_includes/c11-cpp11-and-beyond-and-toolchains.md

```diff
@@ -0,0 +1,179 @@
+### C11/C++11 (and Beyond) and Toolchain Versioning
+
+If your project requires tools compatible with C11, C++11, or a more recent language standard, then it is likely that you will have to upgrade your compiler and/or build tools. This section covers specifically how to upgrade GCC, clang, and cmake; for other dependencies please see [Installing Dependencies](/user/installing-dependencies/).
+
+#### GCC on Linux
+
+Ubuntu 12.04 ships with GCC 4.6.3 and Ubuntu 14.04 ships with GCC 4.8.2.
+
+Note that [GCC support for ISO C11 reached a similar level of completeness as ISO C99 in 4.9](https://gcc.gnu.org/wiki/C11Status) and that C++11 is feature-complete in 4.8.1, but [support for `<regex>` does not exist until 4.9](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53631).
+
+To upgrade GCC to a more recent version, you will have to install the appropriate version from the `ubuntu-toolchain-r-test` source; see below for examples:
+
+```yaml
+matrix:
+  include:
+    # works on Precise and Trusty
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-4.9
+      env:
+         - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+
+    # works on Precise and Trusty
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-5
+      env:
+         - MATRIX_EVAL="CC=gcc-5 && CXX=g++-5"
+
+    # works on Precise and Trusty
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-6
+      env:
+        - MATRIX_EVAL="CC=gcc-6 && CXX=g++-6"
+
+before_install:
+    - eval "${MATRIX_EVAL}"
+```
+
+#### GCC on OS X
+
+```yaml
+matrix:
+  include:
+    - os: osx
+      osx_image: xcode8
+      env:
+        - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+
+    - os: osx
+      osx_image: xcode8
+      env:
+        - MATRIX_EVAL="brew install gcc5 && CC=gcc-5 && CXX=g++-5"
+
+    - os: osx
+      osx_image: xcode8
+      env:
+        - MATRIX_EVAL="brew install gcc && CC=gcc-6 && CXX=g++-6"
```

Comment:
This fails with brew now as it will install gcc 7.2.0 Do you mean gcc6?

In this case the docs need another updating.
```
/Users/travis/.travis/job_stages: line 57: g++-6: command not found

The command "$CXX -O3 -std=c++14 bin2c.cpp -lstdc++fs -o "bin2c"" exited with 127.
```

Reply:
I'm guessing that Homebrew formula is updated to install a new major version.

Reply:
I would presume that yes, you probably have to `brew install gcc-6` or something. Unfortunately I have stuff on my plate right now and don't have the time to come around and fix this - @AraHaan, maybe you could PR a fix?

I think I did try doing `brew install gcc-6` when I submitted this PR, and unfortunately it failed because the latest version was not specifically tagged by homebrew. Perhaps this is also an issue to raise with Homebrew?

Reply:
I think it's ``brew install gcc@6``.

Reply:
Awesome!

Unfortunately I just *know* that I'm going to get notifications in the future when GCC 8 is released and Homebrew points `gcc` to GCC 8 and not 7...